File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    if number < 1:
        raise ValueError("Number must be bigger than 0.")

    aliquot_sum = sum([i for i in range(1, number // 2 + 1) if number % i == 0])

    if aliquot_sum == number:
        return "perfect"

    elif aliquot_sum > number:
        return "abundant"
        
    else:
        return "deficient"

File: javascript/perfect-numbers/perfect-numbers.js
export const classify = (number) => {
  if (number < 1)
    throw new Error('Classification is only possible for natural numbers.');

  const arrayLimit = Math.floor(number / 2) + 1;
  const aliquotSum = [...Array(arrayLimit).keys()].reduce(
    (accumulator, currentValue) =>
      number % currentValue === 0
        ? accumulator + currentValue
        : accumulator
  );

  if (aliquotSum === number)
    return 'perfect';

  else if (aliquotSum > number)
    return 'abundant';

  else
    return 'deficient';
};
